Backpack/Snackbar/Classes/BPKSnackbar.m

```diff
@@ -0,0 +1,369 @@
+            [self.titleLabel.centerYAnchor constraintEqualToAnchor:self.snackbarView.centerYAnchor],
+            [self.snackbarView.trailingAnchor constraintEqualToAnchor:self.actionButton.trailingAnchor constant:BPKSpacingLg],
+            [self.actionButton.centerYAnchor constraintEqualToAnchor:self.snackbarView.centerYAnchor]
+        ]];
+
+        self.snackbarViewTopConstraint = [self.snackbarView.topAnchor constraintEqualToAnchor:self.topAnchor];
+        self.snackbarViewTopConstraint.active = YES;
+
+        self.labelButtonSpacing = [self.actionButton.leadingAnchor constraintGreaterThanOrEqualToAnchor:self.titleLabel.trailingAnchor constant:BPKSpacingBase];
+        self.labelButtonSpacing.active = YES;
+
+        self.labelLeadingSpacing = [self.titleLabel.leadingAnchor constraintEqualToAnchor:self.leftIconContainer.trailingAnchor constant:BPKSpacingMd];
+        self.labelLeadingSpacing.active = YES;
+    }
+
+    return self;
+}
+
+- (void)setUpSnackbarWithTitle:(NSString *)title
+                   buttonTitle:(NSString *_Nullable)buttonTitle
+                    buttonIcon:(UIImage *_Nullable)buttonIcon
+                      leftIcon:(UIImage *_Nullable)leftIcon
+                      duration:(BPKSnackbarDuration)duration
+                viewController:(UIViewController *)viewController
+                      delegate:(id <BPKSnackbarProtocol> _Nullable)delegate
+{
+    self.titleLabel.text = title;
+    self.delegate = delegate;
+    self.snackbarView.backgroundColor = BPKColor.skyBlueShade02;
+    self.parentViewController = viewController;
+    self.actionButton.title = buttonTitle;
+    self.actionButton.image = buttonIcon;
+    self.leftIconContainer.image = leftIcon;
+
+    if (buttonTitle != nil || buttonIcon != nil) {
+        self.actionButton.hidden = NO;
+        self.labelButtonSpacing.constant = BPKSpacingLg;
+        [self.actionButton setContentCompressionResistancePriority:751 forAxis:UILayoutConstraintAxisHorizontal];
+    } else {
+        self.actionButton.hidden = YES;
+        self.labelButtonSpacing.constant = 0;
+        [self.actionButton setContentCompressionResistancePriority:UILayoutPriorityDefaultHigh forAxis:UILayoutConstraintAxisHorizontal];
+    }
+
+    if (leftIcon != nil) {
+        self.leftIconContainer.hidden = NO;
+        self.labelLeadingSpacing.constant = BPKSpacingMd;
+    } else {
+        self.leftIconContainer.hidden = YES;
+        self.labelLeadingSpacing.constant = 0;
+    }
+
+    if (duration != BPKSnackbarDurationIndefinitely) {
+        self.durationInMillis = [self durationInMilliseconds:duration];
+        if (self.durationInMillis > 0) {
+            self.timer = [NSTimer scheduledTimerWithTimeInterval:self.durationInMillis / 1000.0 target:self selector:@selector(snackbarTimeOut) userInfo:nil repeats:NO];
+        }
+    }
+
+    [self updateLayout];
+    [self show];
+}
+
+- (void)updateLayout
+{
```

Comment:
There's a lot of opening braces on new lines. Could you run `clang-format` on the Snackbar code?